### ISSUE TITLE
ObjectNav results on HM3D (test-standard)

### ISSUE THREAD

**Issue by zhangpingrui** [ajzhai/PEANUT#3]
Hello! I'm so sorry to bother you again that I would like to know if the performance of PEANUT on HM3D is test on 500 episodes or on all 2000 episodes? Cause now I use my reproduced checkpoint to obtain a comparable performance when I run 500 episodes, but on all 2000 episodes its performance is lower than before.

**Comment by ajzhai** [comment]
Yea, the ablation study in our paper is on 500 episodes from HM3D val set.

**Comment by zhangpingrui** [comment]
> Yea, the ablation study in our paper is on 500 episodes from HM3D val set.

Ah, so all the results in Table 1(include other methods, such as ProcTHOR) were evaluate on 500 episodes?

**Comment by ajzhai** [comment]
No, that evaluation is using the standard test set on EvalAI: [here](https://eval.ai/web/challenges/challenge-page/1615/overview)

**Comment by zhangpingrui** [comment]
> No, that evaluation is using the standard test set on EvalAI: [here](https://eval.ai/web/challenges/challenge-page/1615/overview)

The results in Table 1 were evaluated online? So the code in nav/collect.py is for ablation? I would like to know whether the code has an evaluation part for the HM3D test standard.

**Comment by zhangpingrui** [comment]
On this website (https://github.com/matterport/habitat-matterport-3dresearch), I really don't find the test-standard dataset of HM3D. There are only train and val. Could you please tell me where to download the test split dataset?

**Comment by ajzhai** [comment]
We cannot download the test dataset. It is intentionally kept hidden from the public so that people cannot optimize their agent's based on test set performance. This is common for many ML benchmarks. The only way to evaluate test set performance is by submitting your docker image to EvalAI. If you want to do that, a few steps need to be done first:

- Change `nav_exp.sh` to contain `python nav/eval.py` (not `collect.py`)
- Uncomment this line in the dockerfile: https://github.com/ajzhai/PEANUT/blob/5fe8358bece89645f7651e524234f967215fa4ac/peanut.Dockerfile#L32
- You need to `docker build` the docker image, but no need for `docker run`
- Follow directions [here](https://eval.ai/web/challenges/challenge-page/1615/submission) to submit.

Note that you should ideally only evaluate the test set performance to report metrics for a paper, not for improving your agent. Use the train and val data to optimize your agent.

**Comment by zhangpingrui** [comment]
> We cannot download the test dataset. It is intentionally kept hidden from the public so that people cannot optimize their agent's based on test set performance. This is common for many ML benchmarks. The only way to evaluate test set performance is by submitting your docker image to EvalAI. If you want to do that, a few steps need to be done first:
> 
> * Change `nav_exp.sh` to contain `python nav/eval.py` (not `collect.py`)
> * Uncomment this line in the dockerfile: https://github.com/ajzhai/PEANUT/blob/5fe8358bece89645f7651e524234f967215fa4ac/peanut.Dockerfile#L32
> * You need to `docker build` the docker image, but no need for `docker run`
> * Follow directions [here](https://eval.ai/web/challenges/challenge-page/1615/submission) to submit.
> 
> Note that you should ideally only evaluate the test set performance to report metrics for a paper, not for improving your agent. Use the train and val data to optimize your agent.

Thanks for your reply! I'll follow your guidance to try to get the metrics!

**Comment by zhangpingrui** [comment]
> We cannot download the test dataset. It is intentionally kept hidden from the public so that people cannot optimize their agent's based on test set performance. This is common for many ML benchmarks. The only way to evaluate test set performance is by submitting your docker image to EvalAI. If you want to do that, a few steps need to be done first:
> 
> * Change `nav_exp.sh` to contain `python nav/eval.py` (not `collect.py`)
> * Uncomment this line in the dockerfile: https://github.com/ajzhai/PEANUT/blob/5fe8358bece89645f7651e524234f967215fa4ac/peanut.Dockerfile#L32
> * You need to `docker build` the docker image, but no need for `docker run`
> * Follow directions [here](https://eval.ai/web/challenges/challenge-page/1615/submission) to submit.
> 
> Note that you should ideally only evaluate the test set performance to report metrics for a paper, not for improving your agent. Use the train and val data to optimize your agent.

Today I tried to do this, but I found that the link to the test-standard summit page was not working.(orz).
![image](https://github.com/ajzhai/PEANUT/assets/74910671/bb395f21-0568-41be-9c87-1c1d9521622d)
If I want to use a metric to justify my approach, what should I do? Can I just use the metrics on HM3D(val)?

**Comment by ajzhai** [comment]
I think you need an account first. Can you make an EvalAI account and try again?

**Comment by zhangpingrui** [comment]
> I think you need an account first. Can you make an EvalAI account and try again?

​Yes, I got that page after I logged in. Maybe you can get in because you have participated in that habitat2022. Can you browse the habitat2021 submit page?

**Comment by ajzhai** [comment]
Yes, you can click on the "Participate" button to register a team.

**Comment by zhangpingrui** [comment]
> Yes, you can click on the "Participate" button to register a team.

Nope, I think you can only create a team during game time. (orz
![image](https://github.com/ajzhai/PEANUT/assets/74910671/89ad2c6b-0d7b-4e96-ad4c-e41e587d6070)

**Comment by ajzhai** [comment]
Hmm that's weird, I thought it was available at any time. You can try asking the Habitat people about that, they can probably help. If there's no solution, then I guess you'll have to use the val split.

**Comment by zhangpingrui** [comment]
> Hmm that's weird, I thought it was available at any time. You can try asking the Habitat people about that, they can probably help. If there's no solution, then I guess you'll have to use the val split.

Good, I'll try. Thanks for your help!
